js/index.js

```diff
@@ -2,30 +2,32 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span').innerText
+  const quantity = product.querySelector('.quantity input').value
+  let subTotal = price * quantity;
 
-  //... your code goes here
+  product.querySelector('.subtotal span').innerText = subTotal;
+  return subTotal
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
   // ITERATION 2
-  //... your code goes here
+  let sum = 0;
+  const products = document.getElementsByClassName('product');
+  for (let product of products) {
+    sum += updateSubtotal(product);
+  }
+
 
   // ITERATION 3
-  //... your code goes here
+  document.querySelector('#total-value span').innerText = sum;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  target.parentElement.parentElement.remove();
 }
```

Comment:
You would need to call the function `calculateAll` at the end of this function to update the total price after removing a product.